Automattic/sensei
# Review all email patterns and the editor configurations

Resolves #6618 
Resolves #6601 
Resolves #6612 


## Proposed Changes
This PR is updating the email template editor to make the Gutenberg editor more similar to the rendered email, simplifying the customization processing and reducing the complexity of maintaining emails. 

It is also enabling the users to customize the background color, margin, and spacing on the templates, previously we are using some hardcoded styles directly on the email template that was overriding the user configurations.
 
**Editor Updates:**
* Hide the post title
* Update the editor background to follow the email background
* Update the Editor area (area to insert the block) to follow the same email sizing
* Some small tweaks on the pattern

In order to reach it I had to update all existent patterns because they were created considering the email template will set some values like font sizes and colors, now it is completely explicitly on the patterns. 


## Testing Instructions
* Enable the feature toggle
* Run `npm run start` to generate the CSS
* Run the email creation tool
* Go to the email template list
* Open each template, except (Lesson Available and Course Expiration)
* Check if the look like the screenshots
* Check if the preview is rendering the emails correctly.


## Screenshots
### Student Emails
* Course Completed  

  <img width="1170" alt="image" src="https://user-images.githubusercontent.com/38718/224388041-c11c9bee-e23d-4385-8fbb-fd3644c5f957.png">
  
* Message Reply Received

  <img width="1167" alt="image" src="https://user-images.githubusercontent.com/38718/224388222-7f3ac87a-9844-421c-b8a6-9247c4301ea1.png">
  
* Quiz Graded

  <img width="1166" alt="image" src="https://user-images.githubusercontent.com/38718/224388372-ba31d838-6641-44fb-a203-888a2e8fb428.png">


----

### Student Emails

* Course Assigned

  <img width="1180" alt="image" src="https://user-images.githubusercontent.com/38718/224389032-be00c41e-7675-46b8-97fb-44156282f4f7.png">

* Course Complete

  <img width="1177" alt="image" src="https://user-images.githubusercontent.com/38718/224389141-0616ca6c-2db2-43e8-b516-90c804a0a1e7.png">


* Course Created

  <img width="1175" alt="image" src="https://user-images.githubusercontent.com/38718/224389332-52b83c9d-46af-4700-9563-3f33014253c6.png">

* Course Started
  
  <img width="1171" alt="image" src="https://user-images.githubusercontent.com/38718/224389475-0f48cfa8-b370-449e-b574-b5219d4ad84a.png">


* Lesson Completed 
  <img width="1177" alt="image" src="https://user-images.githubusercontent.com/38718/224389651-e7ce6b4f-072f-4b61-9cf9-b77c91d90d1e.png">


* Message Reply Received
  <img width="1174" alt="image" src="https://user-images.githubusercontent.com/38718/224389738-ad3963cf-87f3-4e35-a14f-27a0f7f52f5c.png">

* Quiz Submitted
  <img width="1176" alt="image" src="https://user-images.githubusercontent.com/38718/224389959-ffe73471-a12d-4816-9911-ac32a57b512e.png">


* Student send message
  <img width="1174" alt="image" src="https://user-images.githubusercontent.com/38718/224390097-5a24da1d-d843-48a4-a1fd-45469f184252.png">



## Pre-Merge Checklist
